index.js

```diff
@@ -3,26 +3,50 @@
 var path = require('path');
 var extend = require('extend-shallow');
 var isNegated = require('is-negated-glob');
+var isAbsolute = require('is-absolute');
 
 module.exports = function(glob, options) {
+  // shallow clone options
   var opts = extend({}, options);
-  opts.cwd = opts.cwd ? path.resolve(opts.cwd) : process.cwd();
+
+  // ensure cwd is absolute
+  var cwd = path.resolve(opts.cwd ? opts.cwd : process.cwd());
+  if (process.platform === 'win32' || opts.unixify === true) {
+    cwd = unixify(cwd);
+  }
+
+  var rootDir = opts.root;
+  // if `options.root` is defined, ensure it's absolute
+  if (rootDir && !isAbsolute(opts.root)) {
+    if (process.platform === 'win32' || opts.unixify === true) {
+      rootDir = unixify(rootDir);
+    }
+    rootDir = path.resolve(rootDir);
```

Comment:
I've looked into this on a windows vm and this line specifically should be moved above the `if` statement so it matches how `cwd` is handled. Doing this fixes most of the tests on windows.